### PR TITLE
Deck Officer Rename, Warrant Officer Addition

### PR DESCRIPTION
**Pull request by TheGreatRevanchist**
---PRIMARY CHANGES---
--MISC--
Area and object name changes added where necessary (For the Deck Officer)

--RANKS--
Added "Warrant Officers" as a between of enlisted and officers, rank insignias included

-COMMAND-
CMO now able to play from O4 to O5 Fleet and Marines

Deck Chief renamed to Deck Officer, may now able to play from W2 TO W3 and O2 to O3

-SENIOR/SPECIALIST ROLES-
Senior Engineer now able to play from E7 to E8 and W1 to W3
Physician now able to play from O3 to O4
Psychiatrist now able to play from O2 to O3
Brig Chief now able to play from E6 to E7 and W1 to W3
Chaplain now able to play from O2 to O3
Pilot now able to play from E5 to E7 and W1 to W2

I swear to god I've got it now (I hope). If I broke something again I might just commit seppuku with a butter knife.

### CODE REVIEW

File: maps/torch/torch_ranks_boh.dm [hunk 1]
```diff
@@ -78,7 +83,7 @@
 		/datum/species/vasilissan	= list(UNRESTRICTED, /datum/mil_branch/private_security),
 		/datum/species/vulpkanin	= list(UNRESTRICTED, SEMIRESTRICTED),
 		/datum/species/customhuman	= list(UNRESTRICTED, SEMIRESTRICTED),
-		/datum/species/shapeshifter/promethean	= list(UNRESTRICTED, SEMIRESTRICTED),
+		/datum/species/shapeshifter/promethean	= list(/datum/mil_branch/civilian),
```
Comment (Roland410) [inline maps/torch/torch_ranks_boh.dm:86]: This fucks with prometheans again.

File: maps/torch/torch_ranks_boh.dm [hunk 2]
```diff
@@ -90,7 +95,6 @@
 		/datum/species/unathi/yeosa	= list(SMC_TROOPERS_ONLY),
 		/datum/species/humanathi	= list(SMC_TROOPERS_ONLY),
 		/datum/species/tajaran		= list(SMC_TROOPERS_ONLY),
-		/datum/species/shapeshifter/promethean	= list(SMC_TROOPERS_ONLY)
```
Comment (Roland410) [inline maps/torch/torch_ranks_boh.dm:93]: Same (although I wouldn't mind prommie marine heads and such :P).

File: maps/torch/items/clothing/boh_accessory.dm [hunk 3]
```diff
@@ -210,12 +229,46 @@
...
+
+/obj/item/clothing/accessory/solgov/rank/marine_corps/warrant_officer/w1
+	icon_state = "MW1"
+	overlay_state = "armyrank_warrant_officer"
+	name = "ranks (W-1 warrant officer)"
+	desc = "Insignia denoting the rank of Warrant Officer"
+
+/obj/item/clothing/accessory/solgov/rank/marine_corps/warrant_officer/w2
+	icon_state = "MW2"
+	overlay_state = ""
```
Comment (Roland410) [inline maps/torch/items/clothing/boh_accessory.dm:250]: Is this and the rest meant to null out the overlay state, or you forgot to set it?

File: maps/torch/torch_ranks_boh.dm [hunk 4]
```diff
@@ -462,80 +516,109 @@
 	name = "Sergeant Major of the Marine Corps"
 	name_short = "SMMC"
 	accessory = list(/obj/item/clothing/accessory/solgov/rank/marine_corps/enlisted/e9_alt2)
-	sort_order = 9
```
Comment (Roland410) [inline maps/torch/torch_ranks_boh.dm:465]: I believe this was meant to be sort_order = 10?

**Comment by TheGreatRevanchist** [comment]
I'll fix that in a moment, though I'm unsure why a change was made to prometheans. The icon state was me setting it, making a mistake somewhere, having to delete or move something, and not putting it back, I have that set up already otherwise.

Thanks for looking at that for me.

3.5 times a charm

**Comment by Roland410** [comment]
image

**Pull request merged**